File: 0136-single-number/0136-single-number.go
func singleNumber(nums []int) int {

    hashmap := make(map[int]int)
    
    for _, num := range nums {
        hashmap[num]++
    }
    
    for num, count := range hashmap {
        if count == 1 {
            return num
        }
    }
    
    return -1 
}

File: 0136-single-number/0136-single-number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        hashmap = {}

        for num in nums:
            if num in hashmap:
                hashmap[num] += 1
            else:
                hashmap[num] = 1

        for num, count in hashmap.items():
            if count == 1:
                return num
